### ISSUE TITLE
Gamemode Glitch

### ISSUE THREAD

**Issue by ConnorFranta** [EbonJaeger/perworldinventory-kt#74]
1) Go into gamemode creative
![image](https://user-images.githubusercontent.com/26499948/44543632-14fee600-a6de-11e8-8e22-3548a2483499.png)

2) Logout

3) Login (Default is in survival)
![image](https://user-images.githubusercontent.com/26499948/44543662-319b1e00-a6de-11e8-822b-8bc46cd3c510.png)

 - Error occurs with "separate-gamemode-inventories:" & "manage-gamemodes:" set to either true or false

The server is on Spigot 1.12.2

Plugin info:
perworldinventory-kt-2.1.0-legacy

Debug info:
https://pastebin.com/cRjB2D4L

config.yml - https://pastebin.com/WhQpxGR1
worlds.yml - https://pastebin.com/J6m0b1KV
data - https://www.dropbox.com/sh/7fqsmfn9xp3gjqe/AABmgLAWg6_NsMjhR8-mby8Ja?dl=0

**Comment by EbonJaeger** [comment]
Sounds like there is another plugin setting the player's GameMode on join.

**Comment by ConnorFranta** [comment]
I take that back. Found another bug making this unusable...
true, false and false, false on world changes to another group world
🙄 

1.) 
![image](https://user-images.githubusercontent.com/26499948/44546995-a888e480-a6e7-11e8-8907-5c00c6227549.png)




2.) ![image](https://user-images.githubusercontent.com/26499948/44546771-f3562c80-a6e6-11e8-811b-b1f16954852d.png)

3.) 
![image](https://user-images.githubusercontent.com/26499948/44546972-960eab00-a6e7-11e8-8275-e52bdf5e1bf3.png)


Debug:
https://pastebin.com/XfK40aKy

**Comment by EbonJaeger** [comment]
...What is the issue there? To me it looks like things are changing correctly when going to another world. Gonna need more context here.

**Comment by ConnorFranta** [comment]
3rd image is going back to the main world I started in (in the first image) It doesn't save the inventory.

worldFrom='MSEcoSpire', worldTo='BuilderWorld'
Ryohn changed worlds
worldFrom='BuilderWorld', worldTo='MSEcoSpire'
Ryohn changed worlds

**Comment by ConnorFranta** [comment]
> Sounds like there is another plugin setting the player's GameMode on join.

Yes, back to survival. The world is survival and PWI is set as survival.  
You're in gms, you go into /gmc - pick blocks, log out. Login - it sets back to survival, but you have blocks from creative.

**Comment by EbonJaeger** [comment]
I had forgotten that that setting was a thing. Will have to keep this in mind in the future! :D
